cirq-core/cirq/value/condition_test.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+import pytest
+
+import cirq
+
+key_a = cirq.MeasurementKey('a')
+key_b = cirq.MeasurementKey('b')
+init_key_condition = cirq.KeyCondition(key_a)
+init_sympy_condition = cirq.parse_sympy_condition('{a} >= 1')
```

Comment:
Recommend testing around measurement keys containing curly braces, showing both the misbehavior with `parse_sympy_condition` and the workaround with the base constructor.

Reply:
Done